packages/telemetry/src/events/session.ts

```diff
@@ -89,14 +91,23 @@ export function eventCliSession(
 	userConfig?: AstroUserConfig,
 	flags?: Record<string, any>
 ): { eventName: string; payload: EventCliSessionInternal }[] {
+	let optionalIntegrations = 0;
 	const configValues = userConfig
 		? {
 				markdownPlugins: [
 					userConfig?.markdown?.remarkPlugins ?? [],
 					userConfig?.markdown?.rehypePlugins ?? [],
 				].flat(1),
 				adapter: userConfig?.adapter?.name ?? null,
-				integrations: userConfig?.integrations?.map((i: any) => i.name) ?? [],
+				// Filter out falsy integrations
+				integrations: userConfig?.integrations?.filter((i: any) => {
```

Comment:
Mostly just personal preference, but I try to avoid side effects inside of `filter` functions.  Thoughts on moving the filter up with something like this?

```
const integrations = userConfig?.integrations?.filter(Boolean) ?? [];

const configValues = userConfig
  ? {
    // ...
    integrations,
    totalIntegrations: integrations?.length,
    optionalIntegrations: userConfig?.integrations?.length - integrations?.length
  }
```